### ISSUE TITLE
:SetModel causes strange desync when used on player with with certain models

### ISSUE THREAD

**Issue by wrefgtzweve** [Facepunch/garrysmod-issues#6020]
For some reason using :SetModel() on another player clientside causes some kind of desync issue, i'm honestly not sure what to call this.
It happens on p2p x86_64 and srcds linux main branch.

Basegame model example:
```lua
bot:SetModel( "models/props_c17/FurnitureFridge001a.mdl" )
```

Workshop GMA example code:
```lua
steamworks.DownloadUGC( "1982247237", function( path )
    game.MountGMA( path )
    bot:SetModel( "models/player/fillipuster/fillipuster.mdl" )
end)
```
https://github.com/user-attachments/assets/374f2f6d-f7bc-490b-8d67-fb8e8a139045

**Comment by robotboy655** [comment]
This happens because the model is not precached on the server, so when you use the SetModel on client, internally the entity's model index clientside is set to the "invalid index" which causes issues for some reason. This applies to any entity, not just players as far as I can tell, at the very least it happens on prop_physics as well.

**Comment by wrefgtzweve** [comment]
Yeah i can confirm that precaching the models serverside fixes it 👍

**Comment by wrefgtzweve** [comment]
I'm not sure but when looking through this i saw `CModelInfoServer::RegisterDynamicModel` as a possible solution maybe? I don't know how far this works in gmod though and i cant exactly test this myself easily. But maybe some kind of lua binding like `util.PrecacheClientModel` could fix this? Entirely a guess as i cant test it myself.

**Comment by robotboy655** [comment]
It was already tried on clientside only models a couple of years ago and it wasn't working well at all (on x86-64 especially) and was rolled back.

**Comment by wrefgtzweve** [comment]
Might be worth revisiting once x86-64 gets replaced with main branch 64bit 👀
